extensions/wikia/Email/Controller/ForumController.class.php

```diff
@@ -0,0 +1,186 @@
+<?php
+
+namespace Email\Controller;
+
+use Email\Check;
+use Email\EmailController;
+use Email\Fatal;
+
+class ForumController extends EmailController {
+
+	protected $titleText;
+	protected $titleUrl;
+	protected $details;
+	/* @var \Title */
+	protected $board;
+
+	public function initEmail() {
+		$this->titleText = $this->request->getVal( 'titleText' );
+		$this->titleUrl = $this->request->getVal( 'titleUrl' );
+		$this->board = \Title::newFromText(
+			$this->request->getVal( 'boardTitle' ),
+			$this->request->getVal( 'boardNamespace' )
+		);
+		$this->details = $this->getVal( 'details' );
+
+		$this->assertValidParams();
+	}
+
+	protected function assertValidParams() {
+		if ( !$this->board instanceof \Title ) {
+			throw new Check( "Invalid value passed for board" );
+		}
+
+		if ( !$this->board->exists() ) {
+			throw new Check( "Board doesn't exist." );
+		}
+	}
+
+	/**
+	 * @template avatarLayout
+	 */
+	public function body() {
+		$this->response->setData( [
+			'salutation' => $this->getSalutation(),
+			'editorProfilePage' => $this->getCurrentProfilePage(),
+			'editorUserName' => $this->getCurrentUserName(),
+			'editorAvatarURL' => $this->getCurrentAvatarURL(),
+			'summary' => $this->getSummary(),
+			'details' => $this->getDetails(),
+			'detailsHeader' => $this->getDetailsHeader(),
+			'buttonText' => $this->getButtonText(),
+			'buttonLink' => $this->getButtonLink(),
+			'contentFooterMessages' => $this->getContentFooterMessages()
+		] );
+	}
+
+	public function getSubject() {
+		return wfMessage( $this->getSubjectKey(), $this->board->getText() )
+			->inLanguage( $this->targetLang )
+			->text();
+	}
+
+	protected function getSubjectKey() {
+		return 'emailext-forum-subject';
+	}
+
+	protected function getSummary() {
+		return wfMessage( 'emailext-forum-summary', $this->board->getFullURL(), $this->board->getText() )
+			->inLanguage( $this->targetLang )
+			->parse();
+	}
+
+	protected function getDetails() {
+		return $this->details;
+	}
+
+	protected function getButtonText() {
+		return wfMessage( $this->getButtonTextKey() )
+			->inLanguage( $this->targetLang )
+			->parse();
+	}
+	protected function getButtonTextKey() {
+		return 'emailext-forum-button-label';
+	}
+
+	protected function getButtonLink() {
+		return $this->titleUrl;
+	}
+
+	protected function getDetailsHeader() {
+		return $this->titleText;
```

Comment:
`getDetails()`, `getButtonLink()`, and `getDetailsHeader()` are all just returning instance variables inside of the class. Doesn't look like they were overridden by ReplyForumController. Is there a reason to keep these methods? Or can we get rid of them?